File: ToC_2026/8/question4/Haskell/3.hs
//2つの自然数 m と n を取りm < n の真理値を返す lt
iszero :: Int -> Bool
iszero m = m == 0

pi_ :: Int -> Int
pi_ m = if m == 0 then 0 else m - 1

lt :: Int -> Int -> Bool
lt m n = if iszero n
         then False
         else if iszero m
              then True
              else lt (pi_ m) (pi_ n)

main :: IO ()
main = do
    putStrLn "Enter two numbers: "
    [m, n] <- map read . words <$> getLine
    if lt m n
        then putStrLn $ show m ++ " is less than " ++ show n ++ "."
        else putStrLn $ show m ++ " is not less than " ++ show n ++ "."

File: ToC_2026/8/question4/Rust/3.rs
use std::io;

fn iszero(m: i32) -> bool { m == 0 }
fn pi(m: i32) -> i32 { if m == 0 { 0 } else { m - 1 } }

fn lt(m: i32, n: i32) -> bool {
    if iszero(n) {
        false
    } else if iszero(m) {
        true
    } else {
        lt(pi(m), pi(n))
    }
}

fn main() {
    println!("Enter two numbers: ");
    let mut input = String::new();
    io::stdin().read_line(&mut input).unwrap();
    let nums: Vec<i32> = input.split_whitespace().map(|s| s.parse().unwrap()).collect();
    
    if lt(nums[0], nums[1]) {
        println!("{} is less than {}.", nums[0], nums[1]);
    } else {
        println!("{} is not less than {}.", nums[0], nums[1]);
    }
}